internal/featuresupport/features_networking.go

```diff
@@ -348,3 +350,110 @@ func (feature *PlatformManagedNetworkingFeature) getFeatureActiveLevel(cluster *
 
 	return activeLevelNotActive
 }
+
+// SDNNetworkTypeFeature
+type SDNNetworkTypeFeature struct{}
+
+func (feature *SDNNetworkTypeFeature) New() SupportLevelFeature {
+	return &SDNNetworkTypeFeature{}
+}
+
+func (feature *SDNNetworkTypeFeature) getId() models.FeatureSupportLevelID {
+	return models.FeatureSupportLevelIDSDNNETWORKTYPE
+}
+
+func (feature *SDNNetworkTypeFeature) GetName() string {
+	return "Openshift SDN"
+}
+
+func (feature *SDNNetworkTypeFeature) getSupportLevel(filters SupportLevelFilters) models.SupportLevel {
+	// this is for testing, should never be possible in the real world
+	if filters.OpenshiftVersion == "" {
+		return models.SupportLevelSupported
+	}
+	if isAvailable, err := common.BaseVersionLessThan("4.15", filters.OpenshiftVersion); err == nil && isAvailable {
+		return models.SupportLevelSupported
+	}
+	return models.SupportLevelUnavailable
+}
+
+func (feature *SDNNetworkTypeFeature) getFeatureActiveLevel(cluster *common.Cluster, _ *models.InfraEnv, clusterUpdateParams *models.V2ClusterUpdateParams, _ *models.InfraEnvUpdateParams) featureActiveLevel {
+	networkType := getNetworkType(cluster, clusterUpdateParams)
+	if networkType == models.ClusterNetworkTypeOpenShiftSDN {
+		return activeLevelActive
+	}
+	return activeLevelNotActive
+}
+
+func (feature *SDNNetworkTypeFeature) getIncompatibleArchitectures(openshiftVersion *string) *[]models.ArchitectureSupportLevelID {
+	return &[]models.ArchitectureSupportLevelID{}
+}
+
+func (feature *SDNNetworkTypeFeature) getIncompatibleFeatures(string) *[]models.FeatureSupportLevelID {
+	return &[]models.FeatureSupportLevelID{
+		models.FeatureSupportLevelIDOVNNETWORKTYPE,
+	}
+}
+
+func (feature *SDNNetworkTypeFeature) Validate(cluster *common.Cluster, updateParams interface{}) error {
+	if cluster == nil || cluster.OpenshiftVersion == "" {
+		return fmt.Errorf("OpenShift version is not defined")
+	}
+	if isAvailable, err := common.BaseVersionLessThan("4.15", cluster.OpenshiftVersion); err == nil && isAvailable {
+		return nil
+	}
+	return fmt.Errorf("Openshift version %s is not supported for OpenShiftSDN NetworkType", cluster.OpenshiftVersion)
+}
```

Comment:
it's close to `getSupportLevel`, no? Can't we somehow reuse it? I feel like we duplicate validations